frontend/src/pages/auth/components/sign-in-form/sign-in-form.tsx

```diff
@@ -1,16 +1,67 @@
-import { Button } from '#libs/components/components.js';
+import { Button, Input, Link } from '#libs/components/components.js';
+import { AppRoute } from '#libs/enums/enums.js';
+import { useAppForm, useCallback } from '#libs/hooks/hooks.js';
+import {
+  type UserSignInRequestDto,
+  userSignInValidationSchema,
+} from '#packages/users/users.js';
+
+import { DEFAULT_SIGN_IN_PAYLOAD } from './libs/constants.js';
+import styles from './styles.module.scss';
 
 type Properties = {
-  onSubmit: () => void;
+  onSubmit: (payload: UserSignInRequestDto) => void;
 };
 
-const SignInForm: React.FC<Properties> = () => (
-  <>
-    <h1>Sign In</h1>
-    <form>
-      <Button label="Sign in" />
-    </form>
-  </>
-);
+const SignInForm: React.FC<Properties> = ({ onSubmit }) => {
+  const { control, errors, handleSubmit } = useAppForm<UserSignInRequestDto>({
+    defaultValues: DEFAULT_SIGN_IN_PAYLOAD,
+    validationSchema: userSignInValidationSchema,
+  });
+
+  const handleFormSubmit = useCallback(
+    (event: React.BaseSyntheticEvent): void => {
+      void handleSubmit(onSubmit)(event);
+    },
+    [handleSubmit, onSubmit],
+  );
+
+  return (
+    <div className={styles['container']}>
+      <h1 className={styles['title']}>Sign In to your account</h1>
+
+      <form className={styles['form']} onSubmit={handleFormSubmit}>
+        <Input
+          type="text"
+          label="Email"
+          placeholder="Enter your email"
+          name="email"
+          control={control}
+          errors={errors}
+        />
+        <Input
+          type="password"
+          label="Password"
+          placeholder=""
+          name="password"
+          control={control}
+          errors={errors}
+        />
+        <Button
+          type="submit"
+          label="Sign in"
+          className={styles['submit'] as string}
```

Comment:
```suggestion
          className={styles['submit']}
```
Are you sure we need it? In Button component the className property has a default value of empty string

Reply:
@fshabanov I added as string as a temporary solution, because I don't know how to fix the exception we talked about before.
![image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/92298163/4022e063-10a5-4dc4-853d-ff05f458b803)
As I understand, it happens because we use this in tsconfig ([source:](https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes))
```
"exactOptionalPropertyTypes": true
```
Therefore optional parameters won't accept undefined.
Should I leave type string | undefined, leave as string or better solution exists?

Reply:
@Dmytro-Kucherenko let's use 'as string'

Reply:
Already done